### DISCUSSION TITLE
Media ID returned by v1.uploadMedia() but media not uploaded

### DISCUSSION THREAD

**Discussion by t1mp4** [plhery/node-twitter-api-v2#467] [Q&A] [answered]
```typescript
import { TwitterApi, EUploadMimeType } from 'twitter-api-v2';
import fetch from 'node-fetch';

const twitterClient = new TwitterApi({
  appKey: 'nope',
  appSecret: 'nope',
  accessToken: 'nope',
  accessSecret: 'nope',
});

const response = await fetch(
  'https://hlslive-wsczoominwestus.med.nhl.com/editor/3552af25-4e72-4022-910a-cbbabeb4be8d.mp4',
);
const buffer = Buffer.from(await response.arrayBuffer());

twitterClient.v1
  .uploadMedia(buffer, { mimeType: EUploadMimeType.Mp4, longVideo: true })
  .then((val) => {
    console.log(val);
    console.log('success');
  })
  .catch((err) => {
    console.log(err);
  });

```
![Screenshot from 2023-04-25 06-47-52](https://user-images.githubusercontent.com/82466043/234177413-40e98c20-3121-4b45-ae93-35242c55d303.png)

As you can tell by the screenshot, this appears to be working fine. However, despite the success message, the video doesn't actually get uploaded to my Twitter account as expected. Does anyone know possible reasons behind this?

**Comment by 2000JacobHub** [comment, marked as answer]
I think it might be necessary to also use the method v1.tweet()
const tweet = await twitterClient.v1.tweet("", {
        media_ids: media,
    });

**Reply by satyamy846** [reply]
@t1mp4  Yes it is obvious cause uploadMedia is only use to uplaod your media into twitter's server, in order to tweet at your twitter account you need to make POST request to twitter api end point like @2000JacobHub has told you in which you specify the text and media_ids (containing media id which you have uploaded earlier using uploadMedia method)
This will let you upload your video to your twitter account.
